src/main/java/org/openstreetmap/atlas/checks/validation/tag/ConflictingCarAccessibilityCheck.java

```diff
@@ -0,0 +1,202 @@
+import java.util.Optional;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.tags.AccessTag;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.MotorVehicleTag;
+import org.openstreetmap.atlas.tags.MotorcarTag;
+import org.openstreetmap.atlas.tags.VehicleTag;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * This check flags {@link Edge} with conflicting tag combinations resulting from combining
+ * car-navigable/non-car-navigable highway with different modes of transportation tags. This check flags the following tag combinations:
+ * 1) Car-navigable way with restricted car access and with no designated vehicle use tags like {@link org.openstreetmap.atlas.tags.MotorcycleTag}
+ * 2) Car-navigable way with restricted car access and with designated vehicle use tags like {@link org.openstreetmap.atlas.tags.MotorcycleTag}
+ * 3) Non-car-navigable way with open car access
+ * Car accessibility is determined by {@link #checkIfCarAccessible(Edge)}
+ *
+ * @author sayas01
+ */
+public class ConflictingCarAccessibilityCheck extends BaseCheck
+{
+    private static final String CAR_ACCESSIBLE_DESIGNATED_HIGHWAY_INSTRUCTION = "This OSM way  {0,number,#} is designated for specific vehicle uses, consider to add access=NO to prevent general uses.";
+    private static final String CAR_ACCESSIBLE_NON_CAR_NAVIGABLE_INSTRUCTION = "This OSM way  {0,number,#} has a non-car navigable highway tag value combined with an open car access tag value, please verify and make proper corrections if needed.";
+    private static final String CAR_NAVIGABLE_RESTRICTED_ACCESS_HIGHWAY_INSTRUCTION = "This OSM way {0,number,#} has a car navigable highway tag value combined with a restrictive car access tag value, please verify and make proper corrections if needed.";
+    private static final String CONDITIONAL = ":conditional";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(
+            CAR_NAVIGABLE_RESTRICTED_ACCESS_HIGHWAY_INSTRUCTION,
+            CAR_ACCESSIBLE_NON_CAR_NAVIGABLE_INSTRUCTION,
+            CAR_ACCESSIBLE_DESIGNATED_HIGHWAY_INSTRUCTION);
+    private static final String NO = "no";
+    private static final List<String> TAGS_OVERRIDING_ACCESSIBILITY_DEFAULT = Arrays
+            .asList("bus", "minibus", "motorcycle", "taxi", "tourist_bus", "share_taxi", "psv",
+                    "goods", "hsv", "agricultural", "snowmobile", "hov", "public_transportation",
+                    "emergency", "disabled", "hazmat");
+    private static final String YES = "yes";
+    private static final long serialVersionUID = 8896036998080132728L;
+    private final List<String> tagsFilter;
```

Comment:
I did not use "tags.filter" because I do not want the Atlas objects to be filtered by these tags. This list is used to check if any of the items have these tags. If present, then consider for designated use case.

Reply:
Right, and I am not suggesting you use "tages.filter", but you can use the TaggableFilter class to create a filter that uses the same style and won't automatically filter items out. Checkout [SignPostCheck](https://github.com/osmlab/atlas-checks/blob/dev/src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SignPostCheck.java) and/or [InvalidLanesTagCheck](https://github.com/osmlab/atlas-checks/blob/dev/src/main/java/org/openstreetmap/atlas/checks/validation/tag/InvalidLanesTagCheck.java) for examples.

Reply:
Got it! Will refactor the code to use TaggableFilter.

Reply:
This was very helpful! I have refactored the code to use Taggable filter instead of a list of tags. Thanks!